### DISCUSSION TITLE
Sort by: add possibility to specify multiple fields

### DISCUSSION THREAD

**Discussion by erozen** [grafana/grafana#46286] [Dashboards, Panels & Visualizations]
There is a 'Sort by' Transformation, but it will only sort by one item - if you want to sort by column A, then column B, this can't be done.

You can't stack multiple sort-bys, the data gets shuffled before applying the second one.

**Comment by BrunoBlanes** [comment]
# Real use case
I need to create a graph of clients filtered based on a specific criteria per month, but the data comes from two different databases as explained below:

### Graph data
- Columns `Time` and `Quantidade` comes from query `A` to database `Alpha`
- Column `Mes` comes from query `B` to database `Beta`
- Column `monthId` is equivalent in both database `A` and `B`
- Both queries are merged using the built-in `Merge` transformation

![image](https://user-images.githubusercontent.com/28227097/162067025-b4e7fecf-a0e7-497a-942d-27d00ca9804c.png)

Notice how query `A` returned no value for `Janeiro`, therefore that cell is empty and so it is added to the end of the results, which in turn makes the following graph when querying from `now-1y/M` to `now/M-1M`:
![image](https://user-images.githubusercontent.com/28227097/162066931-3e25682c-5e2a-42fd-8516-a199a221a883.png)

`Janeiro` has a `monthId` value of `1`, therefore it should be placed before `Fevereiro`, but since I can only sort by a single column, I can either live with this graph, or sort it by `monthId` which would be 'correct', except it wouldn't be representative of what I'm trying to do with the `now` in the time picker, which is to say I'd like for this to be a sliding graph and not a fixed one.

**Comment by n9** [comment]
@Elfo404 Why this feature request was moved to discussion?

Btw. When I create such transformation via API, the transformation editor will render it. 

![image](https://github.com/grafana/grafana/assets/1651867/e8692cfd-083a-4575-a409-bc57eae27f27)

```json
{
  "id": "sortBy",
  "options": {
    "fields": {}, <-- not sure what should be in this field
    "sort": [
      {
        "desc": false,
        "field": "namespace"
      },
      {
        "desc": true,
        "field": "cluster"
      },
      {
        "desc": true,
        "field": "node"
      },
      {
        "desc": false,
        "field": "pod"
      }
    ]
  }
}
```
However, it is not working.

**Comment by gelicia** [comment]
@n9 This was moved to a discussion, but now we'll be moving it back! You can read more about the move at https://github.com/grafana/grafana/discussions/73424.

The new issue will be tracked at https://github.com/grafana/grafana/issues/77141
